Modules/ZDC/src/ZDCRawDataTask.cxx

```diff
@@ -0,0 +1,1032 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   ZDCRawDataTask.cxx
+/// \author Carlo Puggioni
+///
+
+#include <TCanvas.h>
+#include <TH1.h>
+
+#include "QualityControl/QcInfoLogger.h"
+#include "ZDC/ZDCRawDataTask.h"
+#include <Framework/InputRecord.h>
+#include <Framework/InputRecordWalker.h>
+#include "DPLUtils/DPLRawParser.h"
+#include <TROOT.h>
+#include <TPad.h>
+#include <TString.h>
+#include <TStyle.h>
+#include <TPaveStats.h>
+#include "CommonConstants/LHCConstants.h"
+#include "ZDCSimulation/Digits2Raw.h"
+#include "ZDCSimulation/ZDCSimParam.h"
+#include "FairLogger.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <vector>
+#include <iostream>
+#include <fstream>
+#include <ctype.h>
+#include <boost/algorithm/string.hpp>
+using namespace o2::zdc;
+
+namespace o2::quality_control_modules::zdc
+{
+
+ZDCRawDataTask::~ZDCRawDataTask()
+{
+  delete mHistogram;
+}
+
+void ZDCRawDataTask::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  ILOG(Info, Support) << "initialize ZDCRawDataTask" << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
+
+  // this is how to get access to custom parameters defined in the config file at qc.tasks.<task_name>.taskParameters
+  if (auto param = mCustomParameters.find("myOwnKey"); param != mCustomParameters.end()) {
+    ILOG(Info, Devel) << "Custom parameter - myOwnKey: " << param->second << ENDM;
+  }
+  init();
+  mHistogram = new TH1F("eventSize", "eventSize", 20, 0, 30000);
+  getObjectsManager()->startPublishing(mHistogram);
+  getObjectsManager()->startPublishing(new TH1F("example2", "example2", 20, 0, 30000));
+  try {
+    getObjectsManager()->addMetadata(mHistogram->GetName(), "custom", "34");
+  } catch (...) {
+    // some methods can throw exceptions, it is indicated in their doxygen.
+    // In case it is recoverable, it is recommended to catch them and do something meaningful.
+    // Here we don't care that the metadata was not added and just log the event.
+    ILOG(Warning, Support) << "Metadata could not be added to " << mHistogram->GetName() << ENDM;
+  }
```

Comment:
this is probably some example that might not be needed in the detector QC
```suggestion
```

Reply:
Ok I'll remove it.